src/main/java/Parser.java

```diff
@@ -0,0 +1,93 @@
+import exception.EmptyDescriptionException;
+import exception.EmptyTimeException;
+import exception.UnknownCommandException;
+import exception.UnknownTaskException;
+import exception.UnknownTimeException;
+
+/**
+ * Represents the Parser class, that interprets user input commands.
+ */
+public class Parser {
+
+    /**
+     * Interprets user input text, into understandable commands.
+     *
+     * @param inputText Commands input by user.
+     * @return Command understood by Chatbot.
+     */
+    protected Command commandHandler(String inputText) {
+        try {
+            String[] splitArr = inputText.split(" ");
+            String command = splitArr[0];
+
+            if (command.equals("find")) {
+                String keyword = splitArr[1];
+                return new Command("find", keyword);
+            } else if (command.equals("bye")) {
+                return new Command("bye");
+            } else if (command.equals("list")) {
+                return new Command("list");
+            } else if (command.equals("done")) {
+                if (splitArr.length == 1) {
+                    throw new UnknownTaskException("No task number entered");
```

Comment:
this throwing of error can be abstracted to a subclass DoneCommand (for example), so you can shorten the code for your Parser

Reply:
True! I will do that, thanks :)